### ISSUE TITLE
Create Input Class for feeding calls into NG911 Caller Regions

### ISSUE THREAD

**Issue by jardiamj** [UWB-Biocomputing/Graphitti#370]
An Input class that load calls from an external file and then feeds them to Caller Regions will allow us to test different models of call arrivals (real data, inhomogeneous Poisson process, cluster point process, etc). The idea is that the list of calls can be pre-generated in a more flexible way using a scripting language, without modifications to the Graphitti codebase.

Because Graphitti's architecture is designed to take advantage of parallelization for performance improvement, the Input class should not make calls to directly modify any vertex. Input classes previously implemented in Braingrid got around this constraint by creating a set of `Edges` that are then used to feed the inputs into the corresponding vertices.

We will work on an Input class for the NG911 model and then evaluate if this concept can be extended to the Biological Neural Network one.

**Comment by jardiamj** [comment]
I am using XML as the format for the input files. At the moment it holds a `data` element that contains a list of `event` items each representing an event that occurs at a given vertex. In the case of 911 calls, they have the following attributes:
- vertex_id
- time (time of occurrence)
- duration
- x
- y
- type (one of EMS, Law, or Fire)

Time and duration are expressed as a "clock tick". The size of the clock tick is defined as an attribute of the data element. 

In the case of the Neural Network model, the events are stimuli that occur at a given time on a given vertex so the event attributes are:
- time (of occurrence) and
- vertex_id

An example of an XML input file is down below. I have added a `vertex` attribute that describes the vertex for human readability and a description label of the data for the same reason.

Including a `vertex_id` in the input files ties it to a specific graph (in graphML format), so it might be a good idea to also include the location of the graphML file as part of the input file, for documentation purposes.

```
<?xml version='1.0' encoding='UTF-8'?>
<data description="SPD_calls_sept2020" clock_tick_size="1sec">
  <event>
    <vertex>SEATTLE PD Caller region</vertex>
    <time>0</time>
    <duration>0</duration>
    <x>-122.38496236371942</x>
    <y>47.570236838209546</y>
    <type>EMS</type>
    <vertex_id>194</vertex_id>
  </event>
  <event>
    <vertex>SEATTLE PD Caller region</vertex>
    <time>34</time>
    <duration>230</duration>
    <x>-122.37482094435583</x>
    <y>47.64839548276973</y>
    <type>EMS</type>
    <vertex_id>194</vertex_id>
  </event>
  <event>
    <vertex>SEATTLE PD Caller region</vertex>
    <time>37</time>
    <duration>169</duration>
    <x>-122.4036487601129</x>
    <y>47.55833788618255</y>
    <type>Fire</type>
    <vertex_id>194</vertex_id>
  </event>
...
</data>
```

**Comment by stiber** [comment]
Looks good. Make sure this information gets into the documentation.

**Comment by jardiamj** [comment]
Closed by #394
